solutions/devsprint-firas-safa-1/GitHubApp/Service/Service.swift

```diff
@@ -8,9 +8,22 @@
 import Foundation
 
 struct Service {
-
-    func fetchList(_ completion: ([String]) -> Void) {
-
-        completion(["Repository 1", "Repository 2", "Repository 3"])
+    private let network: NetworkManager
+    
+    init() {
+        self.network = NetworkManager()
```

Comment:
Inject the manager as a parameter, otherwise you can't eventually test stuff. Always inject dependencies.